tests/test_main.py

```diff
@@ -179,6 +181,59 @@ async def test_authenticate_exceptions() -> None:
     assert str(excinfo.value) == "Timeout error: Request timed out"
 
 
+@pytest.mark.parametrize(
+    "fixture_file,method,expected_url,expected_redactions",
+    (
+        (
+            "me.json",
+            "get_homeowner",
+            ME_URL,
+            [
+                '"email": "**redacted**"',
+                '"firstName": "**redacted**"',
+                '"lastName": "**redacted**"',
+                '"deviceSerialNumbers": [\n        "**redacted**"\n    ],\n',
+            ],
+        ),
+        (
+            "notifications.json",
+            "get_notifications",
+            NOTIFICATIONS_URL,
+            [
+                '"serialNumber": "**redacted**"',
+            ],
+        ),
+    ),
+)
+async def test_homeowner_endpoints(
+    caplog: pytest.LogCaptureFixture,
+    fixture_file: str,
+    method: str,
+    expected_url: str,
+    expected_redactions: list[str],
+) -> None:
+    # Create a mock session
+    mock_session = Mock()
+    kem = await get_kem(mock_session)
+    kem.set_timeout(5)
+    mock_response = AsyncMock()
+    mock_response.status = 200
+    mock_response.json.return_value = load_fixture_file(fixture_file)
+    mock_session.get.return_value = mock_response
+
+    with caplog.at_level(logging.DEBUG):
+        caplog.clear()
+        result = await getattr(kem, method)()
+
+    # Assert that the session.get method was called with the correct URL and data
+    mock_session.get.assert_called_once()
+    assert mock_session.get.call_args[0][0] == expected_url
+    assert result == mock_response.json.return_value
+
+    for expected_redaction in expected_redactions:
+        assert expected_redaction in caplog.text
+
+
```

Comment:
Combined these two, to follow the pattern, established with `test_generator_endpoints()`.